### ISSUE TITLE
More robust list comprehension 

### ISSUE THREAD

**Issue by tugsbayasgalan** [pytorch/pytorch#48153]
## 🚀 Feature

We might need to add support for multi dimensional list comprehension and conditional list comprehension
Ideally, we should be able to do something like this:

` class A(nn.Module):
    def __init__(self):
        super().__init__()
        self.x = [[0 for j in range(5)] for i in range(5)]
    def forward(self):
        return self.x[1] 
torch.jit.script(A())`

or 

`class B(nn.Module):
    def __init__(self):
        super().__init__()
        self.x = [0 for i in range(5) if i < 4]
    def forward(self):
        return self.x[1] 
 torch.jit.script(B())
`

## Motivation

List comprehension with conditional statements are pretty common in Python. 

## Pitch

These should work:
` class A(nn.Module):
    def __init__(self):
        super().__init__()
        self.x = [[0 for j in range(5)] for i in range(5)]
    def forward(self):
        return self.x[1] 
torch.jit.script(A())`

or 

`class B(nn.Module):
    def __init__(self):
        super().__init__()
        self.x = [0 for i in range(5) if i < 4]
    def forward(self):
        return self.x[1] 
 torch.jit.script(B())`

## Additional context

This problem is mentioned here: https://github.com/pytorch/pytorch/blob/fcc10d75e1a0c1c2474de559124620ca6b580313/torch/csrc/jit/frontend/tree_views.h#L557.

**Comment by ezyang** [comment]
@zhangguanheng66 wrong triage label; I fixed it

**Comment by ansley** [comment]
I have a PR waiting for review (GH: [47774](https://github.com/pytorch/pytorch/pull/47774)) that implements a basic dict comprehension. Once that's approved, I have plans to follow up by improving the functionality of both list and dict comprehensions since they're pretty similar.

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
